patches/tModLoader/Terraria/ModLoader/ModPlayer.cs

```diff
@@ -1113,4 +1113,16 @@ public virtual IEnumerable<Item> AddStartingItems(bool mediumCoreDeath)
 	public virtual void ModifyStartingInventory(IReadOnlyDictionary<string, List<Item>> itemsByMod, bool mediumCoreDeath)
 	{
 	}
+
+	/// <summary>
+	/// Called when Recipe.FindRecipes is called or the player is crafting an item
+	/// You can use this method to add items as the materials that may be used for crafting items
+	/// </summary>
+	/// <param name="onUsedForCrafting">The action that gets invoked when the item is consumed</param>
+	/// <returns>An list of the items that may be used as crafting materials. If you want to add nothing, return new List&lt;Item&gt;().</returns>
+	public virtual List<Item> AddMaterialsForCrafting(out PlayerLoader.ActionOnUsedForCrafting onUsedForCrafting)
```

Comment:
Please change returntype to `IEnumerable<Item>` and default return value to `null`